include/fastdds/dds/publisher/Publisher.hpp

```diff
@@ -299,11 +283,45 @@ class Publisher : public DomainEntity
             const std::string& profile_name,
             DataWriterQos& qos) const;
 
-    /* TODO
-       bool copy_from_topic_qos(
-            WriterQos& writer_qos,
-            const fastrtps::TopicAttributes& topic_qos) const;
+    /**
+     * @brief Copies TopicQos into the corresponding DataWriterQos
+     * @param[out] writer_qos
+     * @param[in] topic_qos
+     * @return RETCODE_OK if successful, RETCODE_ERROR otherwise
+     */
+    RTPS_DllAPI ReturnCode_t copy_from_topic_qos(
+        fastdds::dds::DataWriterQos& writer_qos,
+        const fastdds::dds::TopicQos& topic_qos) const;
+
+    /**
+     * @brief Deletes all contained DataWriters
+     * @return RETCODE_OK if successful, RETCODE_ERROR otherwise
+     */
+    RTPS_DllAPI ReturnCode_t delete_contained_entities() const;
+
+    /**
+     * @brief Indicates to FastDDS that the contained DataWriters are about to be modified
+     * @return RETCODE_OK if successful, RETCODE_ERROR otherwise
```

Comment:
`RETCODE_ERROR` is not the only error code available in this function. Specify possible Error codes or simply specify "error code" as abstract error kind.